WcaOnRails/app/models/competition.rb

```diff
@@ -466,22 +466,76 @@ def country
   end
 
   def self.search(query, params: {})
-    sql_query = "%#{query}%"
-    Competition.where("id LIKE :sql_query OR name LIKE :sql_query OR cellName LIKE :sql_query OR cityName LIKE :sql_query OR countryId LIKE :sql_query", sql_query: sql_query).order(year: :desc, month: :desc, day: :desc)
+    competitions = Competition.where(showAtAll: true)
```

Comment:
Really? I was under the impression that ActiveRecord doesn't do a SELECT until the last possible moment. Here I am playing around in the rails console (with ActiveRecord logging turn on, so we can see when the SQL is actually run):

```
~/gitting/worldcubeassociation.org/WcaOnRails @kaladin> bin/rails console
Loading development environment (Rails 4.2.1)
irb(main):001:0> ActiveRecord::Base.logger = Logger.new(STDOUT)
=> #<Logger:0x00000004c7a1b0 @progname=nil, @level=0, @default_formatter=#<Logger::Formatter:0x00000004c7a188 @datetime_format=nil>, @formatter=nil, @logdev=#<Logger::LogDevice:0x00000004c7a138 @shift_size=nil, @shift_age=nil, @filename=nil, @dev=#<IO:/dev/null>, @mon_owner=nil, @mon_count=0, @mon_mutex=#<Thread::Mutex:0x00000004c7a0e8>>>
irb(main):002:0> c = Competition.where; nil
=> nil
irb(main):003:0> c
D, [2016-04-12T00:54:57.820164 #23205] DEBUG -- :   Competition Load (4.6ms)  SELECT `Competitions`.* FROM `Competitions`
=> #<ActiveRecord::QueryMethods::WhereChain:0x0000000398b978 @scope=#<ActiveRecord::Relation [#<Competition id: "My0Comp2016", name: "My 0 Best Comp 2016", cityName: "San Francisco", countryId: "Germany", information: "Information!", year: 2016, month: 4, day: 2, endMonth: 4, endDay: 2, eventSpecs: "333 333oh magic", venue: "My backyard", venueAddress: nil, venueDetails: nil, website: "https://www.worldcubeassociation.org", cellName: "My 0 Comp 2016", showAtAll: true, latitude: -23544051, longitude: -69005535, isConfirmed: false, contact: nil, remarks: nil, registration_open: "2016-03-19 02:30:43", registration_close: "2016-03-26 02:30:43", use_wca_registration: true, guests_enabled: true>, #<Competition id: "My100Comp2015", name: "My 100 Best Comp 2015", cityName: "San Francisco", countryId: "Germany", information: "Information!", year: 2015, month: 12, day: 24, endMonth: 12, endDay: 24, eventSpecs: "333 333oh magic", venue: "My backyard", venueAddress: nil, venueDetails: nil, website: "https://www.worldcubeassociation.org", cellName: "My 100 Comp 2015", showAtAll: true, latitude: 33277843, longitude: -102580592, isConfirmed: false, contact: nil, remarks: nil, registration_open: "2016-03-19 02:30:50", registration_close: "2016-03-26 02:30:50", use_wca_registration: true, guests_enabled: true>, #<Competition id: "My101Comp2015", name: "My 101 Best Comp 2015", cityName: "San Francisco", countryId: "USA", information: "Information!", year: 2015, month: 12, day: 23, endMonth: 12, endDay: 23, eventSpecs: "333 333oh magic", venue: "My backyard", venueAddress: nil, venueDetails: nil, website: "https://www.worldcubeassociation.org", cellName: "My 101 Comp 2015", showAtAll: true, latitude: 82346085, longitude: -27666524, isConfirmed: false, contact: nil, remarks: nil, registration_open: "2016-03-19 02:30:50", registration_close: "2016-03-26 02:30:50", use_wca_registration: true, guests_enabled: true>, #<Competition id: "My102Comp2015", name: "My 102 Best Comp 2015", cityName: "San Francisco", countryId: "Germany", information: "Information!", year: 2015, month: 12, day: 22, endMonth: 12, endDay: 22, eventSpecs: "333 333oh magic", venue: "My backyard", venueAddress: nil, venueDetails: nil, website: "https://www.worldcubeassociation.org", cellName: "My 102 Comp 2015", showAtAll: true, latitude: 38482312, longitude: -56055295, isConfirmed: false, contact: nil, remarks: nil, registration_open: "2016-03-19 02:30:50", registration_close: "2016-03-26 02:30:50", use_wca_registration: true, guests_enabled: true>, #<Competition id: "My103Comp2015", name: "My 103 Best Comp 2015", cityName: "San Francisco", countryId: "France", information: "Information!", year: 2015, month: 12, day: 21, endMonth: 12, endDay: 21, eventSpecs: "333 333oh magic", venue: "My backyard", venueAddress: nil, venueDetails: nil, website: "https://www.worldcubeassociation.org", cellName: "My 103 Comp 2015", showAtAll: true, latitude: -48580319, longitude: -126394500, isConfirmed: false, contact: nil, remarks: nil, registration_open: "2016-03-19 02:30:50", registration_close: "2016-03-26 02:30:50", use_wca_registration: true, guests_enabled: true>, #<Competition id: "My104Comp2015", name: "My 104 Best Comp 2015", cityName: "San Francisco", countryId: "Brazil", information: "Information!", year: 2015, month: 12, day: 20, endMonth: 12, endDay: 20, eventSpecs: "333 333oh magic", venue: "My backyard", venueAddress: nil, venueDetails: nil, website: "https://www.worldcubeassociation.org", cellName: "My 104 Comp 2015", showAtAll: true, latitude: -22802675, longitude: -112456437, isConfirmed: false, contact: nil, remarks: nil, registration_open: "2016-03-19 02:30:50", registration_close: "2016-03-26 02:30:50", use_wca_registration: true, guests_enabled: true>, #<Competition id: "My105Comp2015", name: "My 105 Best Comp 2015", cityName: "San Francisco", countryId: "Poland", information: "Information!", year: 2015, month: 12, day: 19, endMonth: 12, endDay: 19, eventSpecs: "333 333oh magic", venue: "My backyard", venueAddress: nil, venueDetails: nil, website: "https://www.worldcubeassociation.org", cellName: "My 105 Comp 2015", showAtAll: true, latitude: -4883678, longitude: 67498992, isConfirmed: false, contact: nil, remarks: nil, registration_open: "2016-03-19 02:30:50", registration_close: "2016-03-26 02:30:50", use_wca_registration: true, guests_enabled: true>, #<Competition id: "My106Comp2015", name: "My 106 Best Comp 2015", cityName: "San Francisco", countryId: "Germany", information: "Information!", year: 2015, month: 12, day: 18, endMonth: 12, endDay: 18, eventSpecs: "333 333oh magic", venue: "My backyard", venueAddress: nil, venueDetails: nil, website: "https://www.worldcubeassociation.org", cellName: "My 106 Comp 2015", showAtAll: true, latitude: -50724679, longitude: -88506667, isConfirmed: false, contact: nil, remarks: nil, registration_open: "2016-03-19 02:30:50", registration_close: "2016-03-26 02:30:50", use_wca_registration: true, guests_enabled: true>, #<Competition id: "My107Comp2015", name: "My 107 Best Comp 2015", cityName: "San Francisco", countryId: "France", information: "Information!", year: 2015, month: 12, day: 17, endMonth: 12, endDay: 17, eventSpecs: "333 333oh magic", venue: "My backyard", venueAddress: nil, venueDetails: nil, website: "https://www.worldcubeassociation.org", cellName: "My 107 Comp 2015", showAtAll: true, latitude: -42491526, longitude: 172288547, isConfirmed: false, contact: nil, remarks: nil, registration_open: "2016-03-19 02:30:50", registration_close: "2016-03-26 02:30:50", use_wca_registration: true, guests_enabled: true>, #<Competition id: "My108Comp2015", name: "My 108 Best Comp 2015", cityName: "San Francisco", countryId: "Brazil", information: "Information!", year: 2015, month: 12, day: 16, endMonth: 12, endDay: 16, eventSpecs: "333 333oh magic", venue: "My backyard", venueAddress: nil, venueDetails: nil, website: "https://www.worldcubeassociation.org", cellName: "My 108 Comp 2015", showAtAll: true, latitude: -14838846, longitude: 158776316, isConfirmed: false, contact: nil, remarks: nil, registration_open: "2016-03-19 02:30:50", registration_close: "2016-03-26 02:30:50", use_wca_registration: true, guests_enabled: true>, ...]>>
irb(main):004:0> c
=> #<ActiveRecord::QueryMethods::WhereChain:0x0000000398b978 @scope=#<ActiveRecord::Relation [#<Competition id: "My0Comp2016", name: "My 0 Best Comp 2016", cityName: "San Francisco", countryId: "Germany", information: "Information!", year: 2016, month: 4, day: 2, endMonth: 4, endDay: 2, eventSpecs: "333 333oh magic", venue: "My backyard", venueAddress: nil, venueDetails: nil, website: "https://www.worldcubeassociation.org", cellName: "My 0 Comp 2016", showAtAll: true, latitude: -23544051, longitude: -69005535, isConfirmed: false, contact: nil, remarks: nil, registration_open: "2016-03-19 02:30:43", registration_close: "2016-03-26 02:30:43", use_wca_registration: true, guests_enabled: true>, #<Competition id: "My100Comp2015", name: "My 100 Best Comp 2015", cityName: "San Francisco", countryId: "Germany", information: "Information!", year: 2015, month: 12, day: 24, endMonth: 12, endDay: 24, eventSpecs: "333 333oh magic", venue: "My backyard", venueAddress: nil, venueDetails: nil, website: "https://www.worldcubeassociation.org", cellName: "My 100 Comp 2015", showAtAll: true, latitude: 33277843, longitude: -102580592, isConfirmed: false, contact: nil, remarks: nil, registration_open: "2016-03-19 02:30:50", registration_close: "2016-03-26 02:30:50", use_wca_registration: true, guests_enabled: true>, #<Competition id: "My101Comp2015", name: "My 101 Best Comp 2015", cityName: "San Francisco", countryId: "USA", information: "Information!", year: 2015, month: 12, day: 23, endMonth: 12, endDay: 23, eventSpecs: "333 333oh magic", venue: "My backyard", venueAddress: nil, venueDetails: nil, website: "https://www.worldcubeassociation.org", cellName: "My 101 Comp 2015", showAtAll: true, latitude: 82346085, longitude: -27666524, isConfirmed: false, contact: nil, remarks: nil, registration_open: "2016-03-19 02:30:50", registration_close: "2016-03-26 02:30:50", use_wca_registration: true, guests_enabled: true>, #<Competition id: "My102Comp2015", name: "My 102 Best Comp 2015", cityName: "San Francisco", countryId: "Germany", information: "Information!", year: 2015, month: 12, day: 22, endMonth: 12, endDay: 22, eventSpecs: "333 333oh magic", venue: "My backyard", venueAddress: nil, venueDetails: nil, website: "https://www.worldcubeassociation.org", cellName: "My 102 Comp 2015", showAtAll: true, latitude: 38482312, longitude: -56055295, isConfirmed: false, contact: nil, remarks: nil, registration_open: "2016-03-19 02:30:50", registration_close: "2016-03-26 02:30:50", use_wca_registration: true, guests_enabled: true>, #<Competition id: "My103Comp2015", name: "My 103 Best Comp 2015", cityName: "San Francisco", countryId: "France", information: "Information!", year: 2015, month: 12, day: 21, endMonth: 12, endDay: 21, eventSpecs: "333 333oh magic", venue: "My backyard", venueAddress: nil, venueDetails: nil, website: "https://www.worldcubeassociation.org", cellName: "My 103 Comp 2015", showAtAll: true, latitude: -48580319, longitude: -126394500, isConfirmed: false, contact: nil, remarks: nil, registration_open: "2016-03-19 02:30:50", registration_close: "2016-03-26 02:30:50", use_wca_registration: true, guests_enabled: true>, #<Competition id: "My104Comp2015", name: "My 104 Best Comp 2015", cityName: "San Francisco", countryId: "Brazil", information: "Information!", year: 2015, month: 12, day: 20, endMonth: 12, endDay: 20, eventSpecs: "333 333oh magic", venue: "My backyard", venueAddress: nil, venueDetails: nil, website: "https://www.worldcubeassociation.org", cellName: "My 104 Comp 2015", showAtAll: true, latitude: -22802675, longitude: -112456437, isConfirmed: false, contact: nil, remarks: nil, registration_open: "2016-03-19 02:30:50", registration_close: "2016-03-26 02:30:50", use_wca_registration: true, guests_enabled: true>, #<Competition id: "My105Comp2015", name: "My 105 Best Comp 2015", cityName: "San Francisco", countryId: "Poland", information: "Information!", year: 2015, month: 12, day: 19, endMonth: 12, endDay: 19, eventSpecs: "333 333oh magic", venue: "My backyard", venueAddress: nil, venueDetails: nil, website: "https://www.worldcubeassociation.org", cellName: "My 105 Comp 2015", showAtAll: true, latitude: -4883678, longitude: 67498992, isConfirmed: false, contact: nil, remarks: nil, registration_open: "2016-03-19 02:30:50", registration_close: "2016-03-26 02:30:50", use_wca_registration: true, guests_enabled: true>, #<Competition id: "My106Comp2015", name: "My 106 Best Comp 2015", cityName: "San Francisco", countryId: "Germany", information: "Information!", year: 2015, month: 12, day: 18, endMonth: 12, endDay: 18, eventSpecs: "333 333oh magic", venue: "My backyard", venueAddress: nil, venueDetails: nil, website: "https://www.worldcubeassociation.org", cellName: "My 106 Comp 2015", showAtAll: true, latitude: -50724679, longitude: -88506667, isConfirmed: false, contact: nil, remarks: nil, registration_open: "2016-03-19 02:30:50", registration_close: "2016-03-26 02:30:50", use_wca_registration: true, guests_enabled: true>, #<Competition id: "My107Comp2015", name: "My 107 Best Comp 2015", cityName: "San Francisco", countryId: "France", information: "Information!", year: 2015, month: 12, day: 17, endMonth: 12, endDay: 17, eventSpecs: "333 333oh magic", venue: "My backyard", venueAddress: nil, venueDetails: nil, website: "https://www.worldcubeassociation.org", cellName: "My 107 Comp 2015", showAtAll: true, latitude: -42491526, longitude: 172288547, isConfirmed: false, contact: nil, remarks: nil, registration_open: "2016-03-19 02:30:50", registration_close: "2016-03-26 02:30:50", use_wca_registration: true, guests_enabled: true>, #<Competition id: "My108Comp2015", name: "My 108 Best Comp 2015", cityName: "San Francisco", countryId: "Brazil", information: "Information!", year: 2015, month: 12, day: 16, endMonth: 12, endDay: 16, eventSpecs: "333 333oh magic", venue: "My backyard", venueAddress: nil, venueDetails: nil, website: "https://www.worldcubeassociation.org", cellName: "My 108 Comp 2015", showAtAll: true, latitude: -14838846, longitude: 158776316, isConfirmed: false, contact: nil, remarks: nil, registration_open: "2016-03-19 02:30:50", registration_close: "2016-03-26 02:30:50", use_wca_registration: true, guests_enabled: true>, ...]>>
```

Notice that the first call to where didn't cause a SELECT, but simple printing c *did* (because the `inspect` method prints out the `@scope=#<ActiveRecord::Relation...`). However, the second time we printed c, nothing was printed.